### ISSUE TITLE
Why the same file exists in asar and asar.unpacked.

### ISSUE THREAD

**Issue by stefanJi** [electron-userland/electron-builder#6949]
* **Electron-Builder Version**: 23.0.3

* **Node Version**:  v14.19.2

* **Electron Version**: 15.3.0
* **Electron Type (current, beta, nightly)**:

* **Target**: win32

```
  "files": [
    "src/lib/*.dll"
  ],
  "asarUnpack": [
    "src/lib/*.dll"
  ]
```

When build finish, *.dll files exists in asar and asar.unpacked. I think this will increment application size.

But If I written config as flow:
```
  "files": [
  ],
  "asarUnpack": [
    "src/lib/*.dll"
  ]
```
*.dll  files will not exsits in asar or asar.unpacked.

So, how to make files only exsits in asar.unpacked ?

**Comment by Jaifroid** [comment]
I have this issue. The `asarUnpack` option merely unpacks files after the asar has been built, instead of preventing them from being added to the asar archive in the first place, so you end up with duplicate files and a larger app than necessary.

**Comment by lukas2** [comment]
I see the same thing on Mac. Normally I think it's just a symbolic link to the unpacked version of my library, but now it's a copy.

**Comment by 10n37** [comment]
Same issue. I really want to migrate to electron-forge from electron-builder, but asarUnpack never worked for me and for the official electron-forge template. 
`asar: { unpack: '**/assets/*' }`
 Sad but true

**Comment by new-carrot** [comment]
They do not exists in two places. When you extract the asar file to check the content, the unpacked files/directories are copied to the extracted output folder.

**Comment by JosXa** [comment]
> They do not exists in two places. When you extract the asar file to check the content, the unpacked files/directories are copied to the extracted output folder.

You're contradicting yourself. I think you mean 
> They **do** exist in two places.

right?

**Comment by new-carrot** [comment]
They do not. The unpack files are in asar.unpack folder only. When we use
'asar extract' command to see the files in asar, the command extract files
in asar and copy the files in unpack filder to the output.

For example, you edit something in unpack file, the extracted output will
have the edit.

Another example we delete the unpack files before extract asar file. The
command will fail because the unpack file is not found.

In summary, this is not a bug. It works as design.

**Comment by new-carrot** [comment]
Yes, it it asar internal logic that keep track the location of the files. If the file is in unpack, asar loads it for you from unpack location. If the file is packed, asar loads it from the asar file.

**Comment by mmaietta** [comment]
>They do not. The unpack files are in asar.unpack folder only. When we use `asar extract` command to see the files in asar, the command extract files in asar and copy the files in unpack filder to the output.

This is my understanding as well.

IIRC, from the asar packaging logic, the asar.unpacked files are simply symlinked into the asar. So when you use `asar extract`, it unpacks both the asar and pulls the symlinked file to the extracted location.

**Comment by finscn** [comment]
> So is there some magic involved that when I attempt to `fs.readFile` an item by its `.../app.asar/...` path, it'll **automatically** rewrite this path be `.../app.asar.unpacked/...`? Or how can we explain that I successfully load files from (what looks like) the packed folder path while the same files also exists in the unpacked folder?

I meet  the same  problem (electron v35)  .  Read a file , sometimes read it in `asar`,  sometime in  `asar.unpacked` .

I think this is an issue .

**Comment by finscn** [comment]
@stefanJi  , Hi , I think you should reopen this issue.
